Added typing hints and type checking across the board + implementation of "RepeatClimateReactAction" feature (#136)
nitaybz/homebridge-sensibo-ac: README.md

> 

Reviewer: Add new line between 63 and 64 above...
usernamePassword
"auth": "usernamePassword",

Author: committed fix.

| `disableHorizontalSwing`   |  When set to `true`, will remove the horizontal swing switch     |          |  `false` |  Boolean |
| `disableVerticalSwing`     |  When set to `true`, will remove the vertical swing control (Oscillate) from the accessory  |          |  `false` |  Boolean |
| `enableClimateReactSwitch` |  Adds a switch to enable/disable Climate React (Smart mode)      |          |  `false` |  Boolean |
| `enableClimateReactAutoSetup` |  When set to `true`, will auto-update the Climate React (Smart mode) settings to match whenever the AC state is set or changed  |          |  `false` |  Boolean  |

> | `climateReactAutoSetupOffset` |  "Temprature offset used when automatically setting Climate React  |          |  `0` |  Number  |

Reviewer: ```suggestion
| `climateReactAutoSetupOffset` |  Offset, in degrees, above or below AC target temperature that will be used to toggle AC when `enableClimateReactAutoSetup` is enabled  |          |  `0` |  Integer  |
```

> | `climateReactAutoSetupMultiplier` |  "Temprature multiplier used when automatically setting Climate React  |          |  `1` |  Number  |

Reviewer: ```suggestion
| `climateReactAutoSetupMultiplier` |  Multiplier above or below AC target temperature that will be used to toggle AC when `enableClimateReactAutoSetup` is enabled  |          |  `1` |  Integer  |
```

> | `enableRepeatClimateReactAction` |  When set to `true`, will Repeat the last Climate React triggered action to decrease the likelihood of AC State discrepancies  |          |  `false` |  Boolean  |

Reviewer: ```suggestion
| `enableRepeatClimateReactAction` |  When set to `true`, will Repeat the last Climate React action triggered, used to decrease the likelihood of AC State discrepancies  |          |  `false` |  Boolean  |
```

> 

Reviewer: Descriptions for climateReactAutoSetupOffset and climateReactAutoSetupMultiplier should also be included.

Author: Accepted the above commits so we're good.

Reviewer: Sorry, I meant a longer description (probably here within Climate React auto setup's description) in addition to the short version in the table.

Author: Got it. Fixed.

To enable **Climate React Auto Setup**, add `"enableClimateReactAutoSetup": true` to your config.

#### Repeat Climate React action

In some scenarios an AC does not receive commands issued by Climate React but the system still "believes" it did. To decrease the likelihood of such a discrepancy persisting, we can repeat (once) the last climate react triggered action, when relevant.

> When enabled, the plugin will periodically check and, when relevant, repeat the last Climate React triggered action to decrease the likelihood of of the scenario described above.

Reviewer: ```suggestion
When enabled, the plugin will periodically check and, when relevant, repeat the last Climate React triggered action to decrease the likelihood of any mismatch.
```
